### PR TITLE
Add nouveau check and modprobe to install

### PR DESCRIPTION
**Pull request by Deezzir**
The PR fixes some issues with the current Nvidia driver install logic:

- There is a check for nouveau drivers; if they are present, tell the operator. Otherwise, there is no point in installing Nvidia's closed-source drivers, as the Nvidia module slot is already occupied
-  Change the order of the arguments passed to `ubuntu-drivers`; see https://bugs.launchpad.net/ubuntu/+source/ubuntu-drivers-common/+bug/2090502 - hilarious
- Run `modprobe` to load the installed drivers, ensuring a reboot is unnecessary. 

The changes were tested on a testflinger machine.

Closes: #361

### CODE REVIEW

**Review by gabrielcocenza — DISMISSED** [review]

File: src/hw_tools.py [hunk 1]
```diff
@@ -277,13 +277,19 @@ def install(self) -> None:
             logger.info("NVIDIA driver already installed in the machine")
             return
 
+        if 0 == subprocess.call("lsmod | grep nouveau", shell=True):
```
[the remark this answers is not in the document]
Comment (Deezzir) [reply src/hw_tools.py:280]: Some

Comment (aieri) [inline src/hw_tools.py:280]: non-blocking: lsmod is just a pretty printer for `/proc/modules`, so I would suggest skipping subprocess and simply reading that file

File: src/hw_tools.py [hunk 2]
```diff
@@ -277,13 +277,19 @@ def install(self) -> None:
...
         logger.info("Installing NVIDIA driver")
         apt.add_package("ubuntu-drivers-common", update_cache=True)
 
         try:
             # This can be changed to check_call and not rely in the output if this is fixed
             # https://github.com/canonical/ubuntu-drivers-common/issues/106
-            result = subprocess.check_output("ubuntu-drivers install --gpgpu".split(), text=True)
+            # https://bugs.launchpad.net/ubuntu/+source/ubuntu-drivers-common/+bug/2090502
+            result = subprocess.check_output("ubuntu-drivers --gpgpu install".split(), text=True)
+            subprocess.call("modprobe nvidia".split())
```
Comment (aieri) [inline src/hw_tools.py:292]: `call` will not throw exceptions for non-zero RCs. Are you sure you don't mean to use `check_call`?

**Review by aieri — CHANGES_REQUESTED** [review]
A couple of suggestions, and we'll need unit tests (but I know you know this :))

File: src/hw_tools.py (hunk 2 above)
Comment (Deezzir) [reply src/hw_tools.py:292]: Yes, it's intentional, I don't want to handle any errors from the modpobe. It should only fail if there is nothing to load or something is already installed, like no nouveau. We handle those cases + plus the charm will check if the driver is loaded by looking at `/proc/driver/nvidia/version `

Comment (Deezzir) [reply src/hw_tools.py:292]: But I guess we can be a little bit more explicit, wdyt

**Review by aieri — APPROVED** [review]

**Review by gabrielcocenza — APPROVED** [review]

**Pull request merged**